vkubelet/pod.go

```diff
@@ -94,25 +98,94 @@ func (s *Server) deletePod(ctx context.Context, namespace, name string) error {
 		return s.forceDeletePodResource(ctx, namespace, name)
 	}
 
-	ctx, span := trace.StartSpan(ctx, "deletePod")
+	ctx, span := trace.StartSpan(ctx, "deletePodFromProvider")
 	defer span.End()
 	ctx = addPodAttributes(ctx, span, pod)
 
-	var delErr error
-	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && errors.IsNotFound(delErr) {
+	if delErr := s.provider.DeletePod(ctx, pod); delErr != nil && errors.IsNotFound(delErr) {
 		span.SetStatus(ocstatus.FromError(delErr))
 		return delErr
 	}
 
 	log.G(ctx).Debug("Deleted pod from provider")
 
-	if !errors.IsNotFound(delErr) {
-		if err := s.forceDeletePodResource(ctx, namespace, name); err != nil {
-			span.SetStatus(ocstatus.FromError(err))
-			return err
-		}
-		log.G(ctx).Info("Deleted pod from Kubernetes")
+	return nil
+}
+
+// terminatePod terminates the pod in the provider. It assumes the API
+// server pod has been marked for deletion
+func (s *Server) terminatePod(ctx context.Context, pod *corev1.Pod) error {
+	ctx, span := trace.StartSpan(ctx, "terminatePod")
+	defer span.End()
+	ctx = addPodAttributes(ctx, span, pod)
+	ctx = span.WithField(ctx, "softDeletes", s.softDeletes)
+
```

Comment:
I suggest adding some logging from this point onwards to increase visibility into what's going on.